var/spack/repos/builtin/packages/py-urllib3/package.py

```diff
@@ -24,21 +26,29 @@ class PyUrllib3(PythonPackage):
     version("1.14", sha256="dd4fb13a4ce50b18338c7e4d665b21fd38632c5d4b1d9f1a1379276bd3c08d37")
 
     variant("socks", default=False, description="SOCKS and HTTP proxy support")
-    variant("secure", default=False, description="Add SSL/TLS support")
+    variant("secure", default=False, description="Add SSL/TLS support", when="@:2.0")
     variant("brotli", default=False, description="Add Brotli support")
+    variant("zstd", default=False, description="Add Zstandard support", when="@2:")
 
     depends_on("python@2.7:2.8,3.4:", when="@:1.25", type=("build", "run"))
     depends_on("python@2.7:2.8,3.5:", when="@1.26.6", type=("build", "run"))
     depends_on("python@2.7:2.8,3.6:3", when="@1.26.12:", type=("build", "run"))
+    depends_on("python@3.7:3", when="@2:", type=("build", "run"))
 
-    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools", when="@1", type="build")
+    depends_on("py-hatchling@1.6.0:1", when="@2", type="build")
 
     depends_on("py-pyopenssl@0.14:", when="+secure")
+    depends_on("py-pyopenssl@17.1.0:", when="+secure @2:")
     depends_on("py-cryptography@1.3.4:", when="+secure")
+    depends_on("py-cryptography@1.9:", when="+secure @2:")
     depends_on("py-idna@2:", when="+secure")
     depends_on("py-certifi", when="+secure")
     depends_on("py-urllib3-secure-extra", when="+secure @1.26.12:")
 
     depends_on("py-pysocks@1.5.6,1.5.8:1", when="+socks")
 
     depends_on("py-brotlipy@0.6:", when="+brotli")
```

Comment:
Only depend on old `py-brotlipy` for version 1.
```suggestion
    depends_on("py-brotlipy@0.6:", when="+brotli @:1")
```